app/assets/javascripts/components/AccountMenu/index.tsx

```diff
@@ -0,0 +1,163 @@
+  appState: AppState;
+  application: WebApplication;
+};
+
+const AccountMenu = observer(({ application, appState }: Props) => {
+  const getProtectionsDisabledUntil = useCallback((): string | null => {
+    const protectionExpiry = application.getProtectionSessionExpiryDate();
+    const now = new Date();
+    if (protectionExpiry > now) {
+      let f: Intl.DateTimeFormat;
+      if (isSameDay(protectionExpiry, now)) {
+        f = new Intl.DateTimeFormat(undefined, {
+          hour: 'numeric',
+          minute: 'numeric'
+        });
+      } else {
+        f = new Intl.DateTimeFormat(undefined, {
+          weekday: 'long',
+          day: 'numeric',
+          month: 'short',
+          hour: 'numeric',
+          minute: 'numeric'
+        });
+      }
+
+      return f.format(protectionExpiry);
+    }
+    return null;
+  }, [application]);
+
+  const [showLogin, setShowLogin] = useState(false);
+  const [showRegister, setShowRegister] = useState(false);
+  const [encryptionStatusString, setEncryptionStatusString] = useState<string | undefined>(undefined);
+  const [isEncryptionEnabled, setIsEncryptionEnabled] = useState(false);
+  const [server, setServer] = useState<string | undefined>(application.getHost());
+  const [isBackupEncrypted, setIsBackupEncrypted] = useState(isEncryptionEnabled);
+  const [protectionsDisabledUntil, setProtectionsDisabledUntil] = useState(getProtectionsDisabledUntil());
+  const [user, setUser] = useState(application.getUser());
+  const [hasProtections] = useState(application.hasProtectionSources());
+
+  const { notesAndTagsCount } = appState.accountMenu;
+
+  const closeAccountMenu = () => {
+    appState.accountMenu.closeAccountMenu();
+  };
+
+  // Add the required event observers
+  useEffect(() => {
+    const removeKeyStatusChangedObserver = application.addEventObserver(
+      async () => {
+        setUser(application.getUser());
+      },
+      ApplicationEvent.KeyStatusChanged
+    );
+
+    const removeProtectionSessionExpiryDateChangedObserver = application.addEventObserver(
+      async () => {
+        setProtectionsDisabledUntil(getProtectionsDisabledUntil());
+      },
+      ApplicationEvent.ProtectionSessionExpiryDateChanged
+    );
+
+    return () => {
+      removeKeyStatusChangedObserver();
+      removeProtectionSessionExpiryDateChangedObserver();
+    };
+  }, [application, getProtectionsDisabledUntil]);
+
+  return (
+    <div className="sn-component">
+      <div id="account-panel" className="sk-panel">
+        <div className="sk-panel-header">
+          <div className="sk-panel-header-title">Account</div>
+          <a className="sk-a info close-button" onClick={closeAccountMenu}>Close</a>
+        </div>
+        <div className="sk-panel-content">
+          <Authentication
+            application={application}
+            server={server}
+            setServer={setServer}
+            closeAccountMenu={closeAccountMenu}
+            notesAndTagsCount={notesAndTagsCount}
```

Comment:
Both of these deal with AppState variables so I think there's no need to pass them down as props, we can just pass appState and read them from state on the child component

Reply:
Correct, done